### PR TITLE
[PIRK-53] - Fix LICENSE and NOTICE Files for Release Compliance

### PR DESCRIPTION
**Pull request by ellisonanne**

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -3,3 +3,276 @@ Copyright 2016 The Apache Software Foundation
...
+
+========================================================================
+Common Development and Distribution License 1.0
+========================================================================
+
+The following components are provided under the Common Development and Distribution License 1.0. See project link for details.
+
+	(Common Development and Distribution License (CDDL) v1.0) JavaBeans Activation Framework (JAF) (javax.activation:activation:1.1 - http://java.sun.com/products/javabeans/jaf/index.jsp)
+    (COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (CDDL) Version 1.0) (GNU General Public Library) Streaming API for XML (javax.xml.stream:stax-api:1.0-2 - no url defined)
+    (CDDL 1.0) Glassfish Jasper (org.mortbay.jetty:jsp-2.1:6.1.14 - http://jetty.mortbay.org/project/modules/jsp-2.1)
```
Comment (smarthi) [inline NOTICE]: Do we have these 2 in Pirk, I think not. Moreover Servlet 2.5 is so 2005.

File: NOTICE [hunk 2]
```diff
@@ -3,3 +3,276 @@ Copyright 2016 The Apache Software Foundation
...
+Elasticsearch for Apache Hadoop (https://github.com/elastic/elasticsearch-hadoop)
+Copyright 2009-2016 The Apache Software Foundation
+
+jna-gmp (https://github.com/square/jna-gmp)
+The Apache Software Foundation
+
+Apache log4j (http://logging.apache.org/log4j/2.x/)
+The Apache Software Foundation
+
+
```
Comment (smarthi) [inline NOTICE]: Need to add one for Jackson

**Comment by smarthi** [comment]
Looks ok to me, but I am no expert in legalese; I would defer to others to comment on this.

**Comment by ellisonanne** [comment]
If you run `mvn license:add-third-party` you will see the dependencies above - if it appeared in the output, I accounted for it in the files...

File: LICENSE [hunk 3]
```diff
@@ -199,4 +199,64 @@
...
+   
+   
+=======================================================================
+Apache Pirk (incubating) Subcomponents:
+
+The Apache Pirk project contains subcomponents with separate copyright
+notices and license terms. Your use of the source code for the these
+subcomponents is subject to the terms and conditions of the following
+licenses.
+
```
Comment (tellison) [inline LICENSE]: Just so there is no misunderstanding here, these LICENSE and NOTICE file changes are only to be included in the exe JAR, that bundles all the 3rd-party dependencies together as a convenience for the end user.

In our single project source JAR (```apache-pirk-x.x.x-incubating-sources.jar```), and Pirk-only binary (```apache-pirk-x.x.x-incubating.jar```), these LICENSE/NOTICE modifications must not appear.  There will be different LICENSE and NOTICE files for the source and binary artefacts.

Ref: http://www.apache.org/dev/licensing-howto.html#bundled-vs-non-bundled

File: LICENSE [hunk 4]
```diff
@@ -199,4 +199,64 @@
...
+
+========================================================================
+BSD-style licenses
+========================================================================
+
+The following components are provided under a BSD-style license. See project link for details.
+The text of each license is also included at licenses/LICENSE-[project].txt.
```
Comment (tellison) [inline LICENSE]: I don't see the ```licenses/``` directory being included in this PR?

We really do need to include the actual text of the 3rd-party dependency licenses in our source distribution.
See  http://www.apache.org/dev/licensing-howto.html#permissive-deps.

It's not quite one for _each_ of the Maven dependencies, things like Scala {compiler, library, parser, xml} are covered by a single license.

For an example, see Spark's [LICENSE](https://github.com/apache/spark/blob/master/LICENSE) and [licenses/ directory](https://github.com/apache/spark/tree/master/licenses).

File: LICENSE [hunk 5]
```diff
@@ -199,4 +199,64 @@
...
+
+The following components are provided under the MIT License. See project link for details.
+The text of each license is also included at licenses/LICENSE-[project].txt.
+
+	 (MIT License) pyrolite (net.razorvine:pyrolite:4.9 - https://github.com/irmen/Pyrolite)
+     (The MIT License) JOpt Simple (net.sf.jopt-simple:jopt-simple:4.6 - http://pholser.github.com/jopt-simple)
+     (MIT License) JCL 1.1.1 implemented over SLF4J (org.slf4j:jcl-over-slf4j:1.7.10 - http://www.slf4j.org)
+     (MIT License) JUL to SLF4J bridge (org.slf4j:jul-to-slf4j:1.7.10 - http://www.slf4j.org)
+     (MIT License) SLF4J API Module (org.slf4j:slf4j-api:1.7.21 - http://www.slf4j.org)
+
```
Comment (tellison) [inline LICENSE]: I've not checked each of these by hand :-) , but trust that they are the set that came from Maven's transitive walk of Pirk's dependencies (```mvn license:add-third-party```).

File: LICENSE (hunk 3 above)
Comment (ellisonanne) [inline LICENSE]: I'm confused - how do we create different LICENSE and NOTICE files for the different jars when they are built via the release plugin?

File: LICENSE (hunk 4 above)
Comment (ellisonanne) [inline LICENSE]: I will add a license directory

File: LICENSE (hunk 5 above)
Comment (ellisonanne) [inline LICENSE]: ....Yes, they came from a (painful :)) transitive walk of the dependencies obtained via `mvn license:add-third-party`

**Comment by tellison** [comment]
Where did the ```licenses/``` directory go?

**Comment by ellisonanne** [comment]
Strange... it's back now.

**Comment by smarthi** [comment]
+1 looks good

**Comment by ellisonanne** [comment]
Just updated pom to exclude licenses during RAT checks

**Comment by smarthi** [comment]
+1 Lgtm

File: pom.xml [hunk 6]
```diff
@@ -350,6 +350,9 @@
 							<exclude>docs/*</exclude> <!-- Exclude docs -->
 							<exclude>logs/*</exclude> <!-- Exclude logs -->
 							<exclude>**/m2.conf</exclude> <!-- Exclude Maven conf which gets installed on travis and fails RAT check -->
+							<exclude>src/main/resources/META-INF/bin-license-notice/licenses/*</exclude> <!-- Exclude licenses files -->
+							<exclude>src/main/resources/META-INF/bin-license-notice/*</exclude>
+							<exclude>src/main/resources/META-INF/*</exclude>
```
Comment (smarthi) [inline pom.xml]: Replace these 3 lines by a single line
`<exclude>src/main/resources/META-INF/**</exclude>`

File: src/main/resources/META-INF/bin-license-notice/NOTICE-bin [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+Elasticsearch for Apache Hadoop (https://github.com/elastic/elasticsearch-hadoop)
+Copyright 2009-2016 The Apache Software Foundation
+
+jna-gmp (https://github.com/square/jna-gmp)
+The Apache Software Foundation
+
+Apache log4j (http://logging.apache.org/log4j/2.x/)
+The Apache Software Foundation
+
+
```
Comment (smarthi) [inline src/main/resources/META-INF/bin-license-notice/NOTICE-bin]: Should we add an entry for Storm, Flink, Beam, Cascading too here?

Comment (ellisonanne) [inline src/main/resources/META-INF/bin-license-notice/NOTICE-bin]: Not until we actually bring them into Pirk

File: pom.xml (hunk 6 above)
Comment (ellisonanne) [inline pom.xml]: will do now

**Comment by smarthi** [comment]
+1 to merge, given that most of the feedback has been addressed now.

**Pull request closed without merging**